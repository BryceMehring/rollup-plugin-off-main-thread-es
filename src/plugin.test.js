const OffMainThread = require('./plugin');
const { rollup } = require('rollup');

const generateAndVerify = async (build, length) => {
  const { output } = await build.generate({
    format: 'es',
  });

  expect(output).toHaveLength(length);

  for (const chunk of output) {
    expect(chunk.code).toMatchSnapshot();
  }
}

describe('plugin', () => {
  it('throw exception if file cannot be found', () => {
    return expect(rollup({
      input: './src/tests/input-bad-worker.js',
      plugins: [
        OffMainThread(),
      ]
    })).rejects.toThrowError();
  });

  it('does nothing when cannot find worker', async () => {
    const build = await rollup({
      input: './src/tests/input-noop.js',
      plugins: [
        OffMainThread(),
      ]
    });

    return generateAndVerify(build, 1);
  });

  it('excludes input files', async () => {
    const build = await rollup({
      input: './src/tests/input-good-worker.js',
      plugins: [
        OffMainThread({
          exclude:  './src/tests/input-good-worker.js',
        }),
      ]
    });

    return generateAndVerify(build, 1);
  })

  it('builds worker', async () => {
    const build = await rollup({
      input: './src/tests/input-good-worker.js',
      plugins: [
        OffMainThread({
          include: './src/tests/input-good-worker.js',
        }),
      ]
    });

    return generateAndVerify(build, 2);
  });
})