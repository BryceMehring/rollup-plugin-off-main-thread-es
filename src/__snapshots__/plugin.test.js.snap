// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugin builds worker 1`] = `
"const test = 1;

const worker = new Worker(new URL('worker-3a9533d0.js', document.baseURI).href, { type: 'module' });

console.log(test);
"
`;

exports[`plugin builds worker 2`] = `
"console.log('hello world from worker');
"
`;

exports[`plugin does nothing when cannot find worker 1`] = `
"console.log('noop');
"
`;

exports[`plugin excludes input files 1`] = `
"const test = 1;

const worker = new Worker('./worker.js', { type: 'module' });

console.log(test);
"
`;
